File: Java/DesignPatterns/src/main/java/designPatterns/structural/composite/Composite.java
package designPatterns.structural.composite;

import java.util.ArrayList;
import java.util.List;

public class Composite implements Component {
  private List<Component> components = new ArrayList<>();

  public void addComponent(Component component) {
    components.add(component);
  }

  public void removeComponent(Component component) {
    components.remove(component);
  }

  @Override
  public String show() {
    StringBuilder stringBuilder = new StringBuilder();
    for (Component component : components) {
      stringBuilder.append(component.show()).append("\n");
    }
    return stringBuilder.toString();
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/structural/composite/Composite.kt
package structural.composite

class Composite : Component {
  private var components: MutableList<Component> = mutableListOf<Component>()

  fun addComponent(component: Component) {
    components.add(component)
  }

  fun removeComponent(component: Component) {
    components.remove(component)
  }

  override fun show(): String {
    val stringBuilder: StringBuilder = java.lang.StringBuilder()
    for (component in components) {
      stringBuilder.append(component.show()).append("\n")
    }
    return stringBuilder.toString()
  }
}